### PR TITLE
Small fix + direct warehouse assignment

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Add assignment module to warehouse to be able to assign couriers directly
- Small fix
-

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/colony/buildings/modules/IAssignmentModuleView.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.minecolonies.api.colony.buildings.HiringMode;
+import com.minecolonies.api.colony.jobs.registry.JobEntry;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * AbstractBuilding View for clients.
+ */
+public interface IAssignmentModuleView extends IBuildingModuleView
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/colony/buildings/modules/IAssignmentModuleView.java:13]: JDoc

File: src/main/java/com/minecolonies/coremod/client/gui/modules/CourierAssignmentModuleWindow.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Assignment module for couriers to warehouse.
+ *
+ */
+public class CourierAssignmentModuleWindow extends AbstractModuleWindow
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/client/gui/modules/CourierAssignmentModuleWindow.java:25]: Jdoc Formatting

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java [hunk 3]
```diff
@@ -101,6 +102,12 @@ public int getModuleMax()
         return building.getBuildingLevel();
     }
 
+    @Override
+    public JobEntry getJobEntry()
+    {
+        return null;
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java]: You sure this is safe like this?
I am pretty sure that method is marked as @NotNull and if so, I would rather this does something else....

**Review by marchermans — CHANGES_REQUESTED** [review]
Formatting and minor nitpickings.

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/CourierAssignmentModule.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import com.minecolonies.api.colony.jobs.registry.JobEntry;
+import com.minecolonies.coremod.colony.jobs.JobDeliveryman;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.FriendlyByteBuf;
+import org.jetbrains.annotations.NotNull;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+/**
+ * The worker module for citizen where they are assigned to if they work at it.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/modules/CourierAssignmentModule.java]: jdoc here aswell^^

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java (hunk 3 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java]: should probably extract the job assignment into a different interface, so we don't require a job on non-working assignment modules

File: src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/PupilBuildingModuleView.java [hunk 5]
```diff
@@ -8,8 +8,8 @@
 public class PupilBuildingModuleView extends WorkerBuildingModuleView
 {
     @Override
-    public boolean canAssign(final ICitizenDataView data)
+    public boolean canAssign(final ICitizenDataView citizen)
     {
-        return data.isChild();
+        return citizen.isChild() && (citizen.getWorkBuilding() == null || getAssignedCitizens().contains(citizen.getId()));
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/PupilBuildingModuleView.java:13]: seems a bit weird, we allow assigning a child which is already assigned?

Comment (MotionlessTrain) [reply src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/PupilBuildingModuleView.java:13]: Isn’t this to show the children in the manage worker GUI? That gui usually shows the unemployed citizens (children) and already assigned workers.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/PupilBuildingModuleView.java:13]: ^

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java]: I added this in here to be able to match them without requiring 100 messages.

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java]: hm not sure I see how this is connected to having 100 messages, care to elaborate?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java]: Basically, the jobEntry is used to decide which IAssignmentModule is called. If it's a job, it's one of the job assignment modules, else it's the livingBuildingModule. I.e. in the warehouse we got couriers. This allows to generalize the modules a little bit and spare us duplicate windows and messages (i.e. more if we create more cases)

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/LivingBuildingModule.java]: I tried to adjust this.

**Review by someaddons — DISMISSED** [review]

**Review by someaddons — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

**Pull request merged**